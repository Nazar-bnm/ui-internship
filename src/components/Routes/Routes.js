import React from 'react';
import { Switch, Route } from 'react-router-dom';
import {
  CSSTransition,
  TransitionGroup
} from 'react-transition-group';

import HomePage from '../HomePage';
import ProductDetailsPageWrapper from '../ProductDetailsPage/ProductDetailsPageWrapper';
import NotFoundPage from '../NotFoundPage';
import DefaultLayoutContainer from '../DefaultLayout/DefaultLayoutContainer';
import WhatIsNew from '../WhatIsNew';
import WishListPageContainer from '../WishlistPage/WishlistPageContainer';
import MostPopular from '../MostPopular';

import './Routes.scss';

const Routes = () => (
  <Route render={({ location }) => (
    <TransitionGroup>
      <CSSTransition
        key={location.key}
        timeout={2000}
        classNames="fade"
      >
        <Switch>
          <DefaultLayoutContainer path="/" exact component={HomePage} />
          <DefaultLayoutContainer path="/home" component={HomePage} />
          <DefaultLayoutContainer path="/product-details" component={ProductDetailsPageWrapper} />
          <DefaultLayoutContainer path="/WhatIsNew" component={WhatIsNew} />
          <DefaultLayoutContainer path="/wishlist" component={WishListPageContainer} />
          <DefaultLayoutContainer path="/MostPopular" component={MostPopular} />
          <DefaultLayoutContainer
            to="/not_found"
            component={NotFoundPage}
            hideFooter
            hideHeader
            hideBrands
            hideShippingInfo
          />
        </Switch>
      </CSSTransition>
    </TransitionGroup>
  )}
  />
);

export default Routes;
