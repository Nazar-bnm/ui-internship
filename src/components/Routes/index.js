import React from 'react';
import { Switch } from 'react-router-dom';

import CounterContainer from '../../containers/Counter';
import HomePage from '../HomePage';
import ProductDetailsPage from '../../containers/ProductDetailsPage';
import ProductListPage from '../ProductListPage';
import NotFoundPage from '../NotFoundPage';
import DefaultLayout from '../DefaultLayout';

const Routs = () => (
  <Switch>
    <DefaultLayout path="/" exact component={HomePage} />
    <DefaultLayout path="/home" component={HomePage} />
    <DefaultLayout path="/products" exact component={ProductListPage} />
    <DefaultLayout path="/products/:id" component={ProductDetailsPage} />
    <DefaultLayout path="/counter" component={CounterContainer} />
    <DefaultLayout
      to="/404"
      component={NotFoundPage}
      hideFooter
      hideHeader
    />
  </Switch>
);

export default Routs;
