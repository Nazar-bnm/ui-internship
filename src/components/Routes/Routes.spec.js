import React from 'react';
import { shallow } from 'enzyme';
import Routes from './Routes';

describe('<NotFoundPage />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Routes />
    );
  });

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
