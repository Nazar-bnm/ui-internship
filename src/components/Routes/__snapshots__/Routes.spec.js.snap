// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage /> should match snapshot 1`] = `
<Switch>
  <DefaultLayout
    component={[Function]}
    exact={true}
    hideBrands={false}
    hideFooter={false}
    hideHeader={false}
    hideShippingInfo={false}
    path="/"
  />
  <DefaultLayout
    component={[Function]}
    hideBrands={false}
    hideFooter={false}
    hideHeader={false}
    hideShippingInfo={false}
    path="/home"
  />
  <DefaultLayout
    component={[Function]}
    hideBrands={false}
    hideFooter={false}
    hideHeader={false}
    hideShippingInfo={false}
    path="/WhatIsNew"
  />
  <DefaultLayout
    component={[Function]}
    hideBrands={false}
    hideFooter={false}
    hideHeader={false}
    hideShippingInfo={false}
    path="/MostPopular"
  />
  <DefaultLayout
    component={[Function]}
    hideBrands={true}
    hideFooter={true}
    hideHeader={true}
    hideShippingInfo={true}
    to="/not_found"
  />
</Switch>
`;
